wallet/lib/pages/purchase_item/purchase_item_view_model.dart

```diff
@@ -454,15 +435,49 @@ class PurchaseItemViewModel extends ChangeNotifier {
     return const Right(true);
   }
 
-
   void addLogForCart() {
+    repository.logAddToCart(
+      recipeId: nft.recipeID,
+      recipeName: nft.name,
+      author: nft.creator,
+      purchasePrice: double.parse(nft.price) / kBigIntBase,
+      currency: nft.ibcCoins.name,
+    );
+  }
 
-    repository.logAddToCart(recipeId: nft.recipeID, recipeName: nft.name, author: nft.creator, purchasePrice: double.parse(nft.price) / kBigIntBase, currency: nft.ibcCoins.name, );
+  @visibleForTesting
+  bool isIOSStripePaymentAllowed({required bool isPlatformAndroid}) {
+    if (isPlatformAndroid) {
+      return true;
+    } else {
+      return nft.iosStripePaymentAllowed;
+    }
+  }
+
+  /// Conditions
+  /// If item is available for buying
+  /// If nft is free drop show the button
+  /// If any currency other than stripe usd show button
+  /// If stripe payment is allowed or nft
+  bool showBuyNowButton({required bool isPlatformAndroid}) {
+    if (!(nft.amountMinted < nft.quantity)) {
+      return false;
+    }
+
+    if (double.parse(nft.price) == 0) {
+      return true;
+    }
+
+    if (nft.ibcCoins != IBCCoins.ustripeusd) {
+      return true;
+    }
+
+    return isIOSStripePaymentAllowed(isPlatformAndroid: isPlatformAndroid);
   }
 
   NFT get nft => _nft;
 
-  final ValueNotifier<bool> shouldShowBuyNow = ValueNotifier(false);
+  // final ValueNotifier<bool> shouldShowBuyNow = ValueNotifier(false);
```

Comment:
I think this dead code can be removed as well.

Reply:
yes sure.